### PR TITLE
feat: LT-1185 - Goods flags

### PR DESCRIPTION
**Pull request by cstobart**

### CODE REVIEW

**Comment by efahy** [comment]
I currently get this error in the API output when I try to assign a flag to a good: `Not Found: /goods/3265ed59-4ebd-4632-8e7c-6fe648a44be6/flags/`

**Comment by efahy** [comment]
The scenarios in the story talk about "When I select one or more goods" - I can't see in the UI how to select multiple goods, currently.  Is this something you're still working on, maybe?

**Comment by cstobart** [comment]
Currently the frontend and backend aren't synced as the frontend only handles single goods edits whereas the backend expects an array, going to fix this later

**Comment by efahy** [comment]
The story is not clear on whether goods type things on open licenses should be "flaggable" but I see this is not possible currently - have you checked what Anne's expectations are are around this?

File: assets/javascripts/edit-good-flags.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+var buttonGiveAdvice = $("#button-edit-goods-flags")
+
+function setEditGoodsFlagsAbility() {
+	// Disable the give advice button unless a checkbox is selected
```
Comment (jfaracik) [inline assets/javascripts/edit-good-flags.js]: incorrect comment

File: cases/services.py [hunk 2]
```diff
@@ -157,3 +157,38 @@ def get_ecju_queries(request, pk):
...
+
+
+def get_good_activity(request, pk):
+    data = get(request, GOOD_URL + pk + ACTIVITY_URL)
+    return data.json(), data.status_code
+
+
+# Good Flags
+# Always takes an array of good id's
+def put_good_flags(request, json):
```
Comment (jfaracik) [inline cases/services.py]: Delete

File: cases/views/flags.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    form = None
+    selected_flags = None
+    url = None
+    level = None
+
+    def dispatch(self, request, *args, **kwargs):
+        case_id = str(kwargs['pk'])
+        kwargs = {'pk': case_id}
+        self.level = request.GET.get('level')
+        self.objects = request.GET.getlist(
```
Comment (jfaracik) [inline cases/views/flags.py]: formatting

File: conf/constants.py [hunk 4]
```diff
@@ -22,7 +23,10 @@
 GOV_USERS_ROLES_URL = '/gov-users/roles/'
 GOV_USERS_PERMISSIONS_URL = '/gov-users/permissions/'
 FLAGS_URL = '/flags/'
+ASSIGN_FLAGS_URL = FLAGS_URL + 'assign/'
+GOODS_FLAGS_URL = 'flags/'
 FLAGS_CASE_LEVEL_FOR_TEAM = '/flags/?level=Case&team=True'
+FLAGS_GOOD_LEVEL_FOR_TEAM = '/flags/?level=Good&team=True'
```
Comment (jfaracik) [inline conf/constants.py]: Team is a really bad name for this

File: templates/cases/case/good.html [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    {% if good.documents %}
+{#        <p class="govuk-caption-m">{% get_string 'goods.documents.description' %}</p>#}
+        {% for document in good.documents %}
+            <div class="lite-card lite-card--download">
+                <div>
+                    <p class="govuk-body govuk-!-font-weight-bold" style="margin-bottom: 10px;">{{ document.name }}</p>
+                    <p class="govuk-body" style="margin-bottom: 10px">{{ document.description|slice:"0:40" }} {% if document.description|length > 40 %}...{% endif %}</p>
+                </div>
+                <div class="lite-vertical-align">
+{#                    {% if document.safe == True %}#}
```
Comment (jfaracik) [inline templates/cases/case/good.html]: Why is this commented out?

File: templates/cases/case/views/case-flags.html [hunk 6]
```diff
@@ -10,18 +10,18 @@
...
 	<div class="govuk-form-group">
-		<a href="{% url 'cases:assign_flags' case.id %}" id="application-edit-case-flags" class="govuk-link govuk-link--no-visited-state">
+		<a href="{% url 'cases:assign_flags' case.id %}?level=cases&cases={{ case.id }}" id="application-edit-case-flags" class="govuk-link govuk-link--no-visited-state">
 			{% get_string 'cases.case.edit_case_flags' %}
 		</a>
 	</div>
 {% else %}
 	<div class="govuk-form-group">
-		<a href="{% url 'cases:assign_flags' case.id %}" id="application-edit-case-flags" class="govuk-link govuk-link--no-visited-state">
+		<a href="{% url 'cases:assign_flags' case.id %}?level=cases&cases={{ case.id }}" id="application-edit-case-flags" class="govuk-link govuk-link--no-visited-state">
```
Comment (jfaracik) [inline templates/cases/case/views/case-flags.html]: Could the level not be inferred from cases:assign flags?

Comment (cstobart) [reply templates/cases/case/views/case-flags.html]: nope

File: ui_automation_tests/pages/queues_pages.py [hunk 7]
```diff
@@ -20,7 +20,13 @@ def click_add_a_queue_button(self):
...
         self.driver.set_timeout_to(0)
         no = len(self.driver.find_elements_by_link_text(app_id))
-        self.driver.set_timeout_to_10_seconds()
+        url = self.driver.current_url
+        page_number = 1
+        while no == 0:
+            page_number += 1
+            next_page = url + '&page=' + str(page_number)
+            self.driver.get(next_page)
+            no = len(self.driver.find_elements_by_link_text(app_id))
```
[the remark this answers is not in the document]
Comment (cstobart) [reply ui_automation_tests/pages/queues_pages.py]: different method, only use case

File: cases/services.py [hunk 8]
```diff
@@ -157,3 +157,41 @@ def get_ecju_queries(request, pk):
...
+
+
+# Good Flags
+# Always takes an array of good id's
+def get_flags_for_team_of_level(request, level):
+    data = get(request, FLAGS_URL + '?level=' + level + '&team=True')
+    return data.json(), data.status_code
+
+
+def get_object(request, level, pk):
```
Comment (jfaracik) [inline cases/services.py]: get_flagged_object -> not get_object

File: cases/views/main.py [hunk 9]
```diff
@@ -240,38 +251,39 @@ def _handle_ecju_query_form_errors(self, case_id, ecju_query, request):
...
-                error_list = []
-                for key in errors:
-                    error_list.append("{field}: {error}".format(field=key, error=errors[key][0]))
-                error = "\n".join(error_list)
-            return error_page(request, error)
-
-        return redirect('/cases/clc-query/' + case_id + '#case_notes')
+#
+# class ViewCLCCase(TemplateView):
+#     def get(self, request, **kwargs):
```
Comment (jfaracik) [inline cases/views/main.py]: Why is this all commented out?

File: cases/views/main.py [hunk 10]
```diff
@@ -43,28 +44,32 @@ def get(self, request, **kwargs):
...
-
-        # Add assigned users to each case
-        queue['queue']['cases'] = add_assigned_users_to_cases(queue['queue']['cases'],
-                                                              case_assignments['case_assignments'])
-
-        # Get current query parameters to inject into template for sorting with filters
-        current_filter_url = request.GET.urlencode().split('&')
+        # case_assignments, status_code = get_queue_case_assignments(request, queue_id)
+        #
+        # # Add assigned users to each case
```
Comment (jfaracik) [inline cases/views/main.py]: Commented out?

File: cases/services.py [hunk 11]
```diff
@@ -157,3 +157,30 @@ def get_ecju_queries(request, pk):
...
+
+
+# Good Flags
+# Always takes an array of good id's
+def get_flags_for_team_of_level(request, level):
+    data = get(request, FLAGS_URL + '?level=' + level + '&team=True')
+    return data.json(), data.status_code
+
+
+def put_objects_flags(request, json):
```
Comment (jfaracik) [inline cases/services.py]: Rename put_objects_flags to put_flags

File: cases/services.py [hunk 12]
```diff
@@ -157,3 +157,30 @@ def get_ecju_queries(request, pk):
...
+
+
+def get_good_activity(request, pk):
+    data = get(request, GOOD_URL + pk + ACTIVITY_URL)
+    return data.json(), data.status_code
+
+
+# Good Flags
+# Always takes an array of good id's
+def get_flags_for_team_of_level(request, level):
```
Comment (jfaracik) [inline cases/services.py]: Use Typing not a comment above the method https://docs.python.org/3/library/typing.html

**Pull request merged**